### PR TITLE
[MER-1672] Manage page purpose UI

### PR DESCRIPTION
**Pull request by augusto1024**
### NOTES
This PR adds a `Hierarchy Selector` component made with Surface and it's used for selecting the related resources of an exploration page. Here's what it looks like:

https://user-images.githubusercontent.com/53446634/217830041-79417c0e-633e-478d-b744-fed462177853.mov

I know we have a `Hierarchy Picker` component but it's not very generic and it didn't feel right for this use case.

### CODE REVIEW

File: lib/oli_web/components/hierarchy_selector.ex [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        assign(
+          socket,
+          selected_values: selected_values
+        )
+
+      {:noreply, socket}
+    end
+  end
+
+  defp input_name(form_or_name, field)
```
Comment (simonchoxx) [inline lib/oli_web/components/hierarchy_selector.ex:166]: Just to understand... I'm not sure what this function returns here.

File: lib/oli_web/live/curriculum/container/container_live.ex [hunk 2]
```diff
@@ -86,21 +89,28 @@ defmodule OliWeb.Curriculum.ContainerLive do
            children: children,
            active: :curriculum,
            breadcrumbs:
-             Breadcrumb.trail_to(project_slug, container.slug, Oli.Publishing.AuthoringResolver, project.customizations),
+             Breadcrumb.trail_to(
+               project_slug,
+               container.slug,
+               Oli.Publishing.AuthoringResolver,
+               project.customizations
+             ),
```
Comment (simonchoxx) [inline lib/oli_web/live/curriculum/container/container_live.ex:97]: This here was done by the formatter maybe? 😅

**Review by simonchoxx — COMMENTED** [review]
Awesome work mate 🚀🚀!!! I just left you some annotations, feel free to look at them if you want!

File: lib/oli_web/components/hierarchy_selector.ex [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+  def handle_event("expand", _params, socket) do
+    if socket.assigns.expanded do
+      socket = assign(socket, expanded: false)
+      {:noreply, socket}
+    else
+      socket = assign(socket, expanded: true)
+      {:noreply, socket}
+    end
+  end
```
[the remark this answers is not in the document]
Comment (augusto1024) [reply lib/oli_web/components/hierarchy_selector.ex:136]: 100% agree

File: lib/oli_web/components/hierarchy_selector.ex (hunk 1 above)
Comment (augusto1024) [reply lib/oli_web/components/hierarchy_selector.ex:166]: Nothing really, I just stole this from `Phoenix.HTML.Form` 😅

File: lib/oli_web/live/curriculum/container/container_live.ex (hunk 2 above)
Comment (augusto1024) [reply lib/oli_web/live/curriculum/container/container_live.ex:97]: Yes, I ran `mix format` in all the files I changed

**Review by darrensiegel — APPROVED** [review]

**Pull request merged**